### PR TITLE
Content moderation, format/size normalization and code generalization for image uploads

### PR DESCRIPTION
**Pull request by cmyui**
Add content moderation using AWS rekognition, content normalization (in format and sizes), and also begin to generalize this service to act as the central user content management service.

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -15,10 +15,15 @@ def read_bool(value: str) -> bool:
...
+S3_AWS_REGION = os.environ["AWS_REGION"]
+S3_AWS_ACCESS_KEY_ID = os.environ["AWS_ACCESS_KEY_ID"]
+S3_AWS_SECRET_ACCESS_KEY = os.environ["AWS_SECRET_ACCESS_KEY"]
+S3_AWS_ENDPOINT_URL = os.environ["AWS_ENDPOINT_URL"]
+S3_AWS_BUCKET_NAME = os.environ["AWS_BUCKET_NAME"]
+
+REKOGNITION_AWS_REGION = os.environ["REKOGNITION_AWS_REGION"]
+REKOGNITION_AWS_ACCESS_KEY_ID = os.environ["REKOGNITION_AWS_ACCESS_KEY_ID"]
+REKOGNITION_AWS_SECRET_ACCESS_KEY = os.environ["REKOGNITION_AWS_SECRET_ACCESS_KEY"]
+REKOGNITION_AWS_ENDPOINT_URL = os.environ["REKOGNITION_AWS_ENDPOINT_URL"]
```
Comment (cmyui) [inline settings.py]: Still needs additions in vault before rollout

Comment (cmyui) [reply settings.py]: Also, I know aws services have a commonality of often depending on specific env variables directly, some of which we previously used:

```py
AWS_REGION = os.environ["AWS_REGION"]
AWS_ACCESS_KEY_ID = os.environ["AWS_ACCESS_KEY_ID"]
AWS_SECRET_ACCESS_KEY = os.environ["AWS_SECRET_ACCESS_KEY"]
AWS_ENDPOINT_URL = os.environ["AWS_ENDPOINT_URL"]
AWS_BUCKET_NAME = os.environ["AWS_BUCKET_NAME"]
```

We will want to be careful during the rollout to identify any cases where we were previously depending on this behaviour -- there may be breaking changes.

File: app/usecases/images.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+from app.adapters import s3
+from app.errors import Error
+from app.errors import ErrorCode
+
+DISALLOWED_MODERATION_LABELS = [
+    "Explicit Nudity",
+    "Suggestive",
+    "Violence",
+    "Visually Disturbing",
+]
```
Comment (cmyui) [inline app/usecases/images.py:35]: Going with a whitelist approach here, seems a bit safer

Comment (cmyui) [reply app/usecases/images.py:35]: This list is not yet finalized, will do another round of confirming we have all the desirable behaviour before merge

Comment (cmyui) [reply app/usecases/images.py:35]: Here are the labels: https://docs.aws.amazon.com/rekognition/latest/dg/moderation.html#moderation-api

File: app/api/v1/avatars.py [hunk 3]
```diff
@@ -1,12 +1,49 @@
...
+        image_type=ImageType.USER_AVATAR,
+        image_content=file_content,
+        no_ext_file_name=f"{user_id}",
+    )
+    if isinstance(data, Error):
+        return Response(
+            status_code=_get_status_code_for_error(data.code),
+            content=data.user_feedback,
+        )
+    return Response(status_code=201)
```
Comment (cmyui) [inline app/api/v1/avatars.py:44]: Still need to implement the upload endpoints for other resources

File: settings.py (hunk 1 above)
Comment (cmyui) [reply settings.py]: All of the new env vars have been added. The removed ones will stay until after the safe deployment.

**Review by infernalfire72 — DISMISSED** [review]

File: settings.py [hunk 4]
```diff
@@ -15,10 +15,15 @@ def read_bool(value: str) -> bool:
...
-AWS_REGION = os.environ["AWS_REGION"]
-AWS_ACCESS_KEY_ID = os.environ["AWS_ACCESS_KEY_ID"]
-AWS_SECRET_ACCESS_KEY = os.environ["AWS_SECRET_ACCESS_KEY"]
-AWS_ENDPOINT_URL = os.environ["AWS_ENDPOINT_URL"]
-AWS_BUCKET_NAME = os.environ["AWS_BUCKET_NAME"]
+AWS_S3_REGION = os.environ["AWS_S3_REGION"]
+AWS_S3_ACCESS_KEY_ID = os.environ["AWS_S3_ACCESS_KEY_ID"]
+AWS_S3_SECRET_ACCESS_KEY = os.environ["AWS_S3_SECRET_ACCESS_KEY"]
+AWS_S3_ENDPOINT_URL = os.environ["AWS_S3_ENDPOINT_URL"]
+AWS_S3_BUCKET_NAME = os.environ["AWS_S3_BUCKET_NAME"]
```
Comment (tsunyoku) [inline settings.py:22]: make sure these are renamed in vault

File: app/api/v1/avatars.py [hunk 5]
```diff
@@ -1,12 +1,49 @@
...
+        }[error_code]
+    except KeyError:
+        logging.warning(
+            "Unmapped error code while resolving http status code",
+            extra={"error_code": error_code},
+        )
+        return 500
+
+
+@router.post("/api/v1/users/{user_id}/avatar")
```
Comment (cmyui) [inline app/api/v1/avatars.py]:
```suggestion
# @router.post("/api/v1/users/{user_id}/avatar")
```

File: settings.py (hunk 4 above)
Comment (cmyui) [reply settings.py:22]: Done

File: app/api/v1/avatars.py (hunk 3 above)
Comment (cmyui) [reply app/api/v1/avatars.py:44]: Will come in a later PR

File: settings.py (hunk 1 above)
Comment (cmyui) [reply settings.py]: > We will want to be careful during the rollout to identify any cases where we were previously depending on this behaviour -- there may be breaking changes.

On second thought this is not a concern as we will not use these APIs to begin with. We can safely remove the old vars before anything will call this.

**Pull request merged**